clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/client/Client.kt

```diff
@@ -192,14 +192,16 @@ class Client(
     fun waitForServerRequestsToFinish(timeout: Long = SERVER_TIMEOUT) {
         runBlocking {
             withTimeout(timeout) {
-                while (requestsCS.hasChildren()) {
+                val hasChildJobs = requestsCS.coroutineContext.job.children.any()
+                while (hasChildJobs) {
                     delay(DELAY_TIME)
                 }
             }
         }
     }
 
     companion object {
+        //TODO: why don't use process this setting similarly to "isFirstLaunch"? Why is it a property of the client?
```

Comment:
There is no need to store this setting in file system (as we do for `isFirstLaunch`). This setting is changed in tests before starting the plugin for tests. Then when plugin launches, we check this setting in `StartupActivity`. A better a way would be not use this "global" variable, but ask from IDE whether it was launched in test mode or not

Reply:
I found `Application.isUnitTestMode`, so this variable is not needed. You can remove it, or leave this todo and I will fix that

Reply:
Ok. I removed TODO.